.github/pull_request_template.md

```diff
@@ -0,0 +1,21 @@
+## Describe the problem
+
+## Describe the solution
+
+## Risk & Testing considerations
+What is the risk level of this change (Low / Medium / High)? For Medium / High please elaborate.
+
+Which modules are impacted by this change?
+
+Which integration tests and or unit tests are applicable for it? This could be a whole suite of Python tests or specific tests.
+
+Has new tests been written which exercises the relevant code paths? If yes, please identify this via the code coverage report. If not, explain why.
+
+What other testing is relevant for this change? This could include: stress testing, MongoDB patch builds, etc.
+
```

Comment:
Like my comment on the first bit, I wonder if theres too much structure here to the point where it will discourage people from wanting to introduce the template.

What would it look like if this was 

```suggestion
## What makes this change safe?
<!-- DELETE WHEN COMPLETING
A good answer to this question helps the reviewer understand where they should focus their attention, consider these questions:
- Is the change risky or not? Why?
- What tests are you adding? Why?
- What existing tests are you relying on? 
- What are you worried about that you'd like the reviewer to focus on?
```

Reply:
I agree making the template succinct would be important for the initial introduction. 
@luke-pearson as you are the main author of this section of contents, I'd like to hear your opinions.

Reply:
I agree with this, i believe the pull request has too much structure here. I feel like Jonas goes in more of a better direction, having a more looser structure gives more room for developers to breathe. 

We should still set a risk level for each change. Developers usually have a gauge already with their confidence levels on their changes. It would also help drop barons, reviewers and give the team a level of confidence around the what changes could produce fallout.

Reply:
That makes sense. My feeling is that it's probably better to go for a simpl(ish) version to start with, and then add more detail later as needed. I suspect that there is less risk of pushback.

Reply:
I think the simpler version is good, but this question seems a bit too honest to me:
- What are you worried about that you'd like the reviewer to focus on?

Maybe given the context of our PRs we should think of something like:
- What sections of the change should the reviewer take extra care reviewing?

Reply:
@luke-pearson I can see re-wording it in that direction, but I can also see that being a good discussion point for how to think about review.

The subtle difference between these two questions might not matter. The latter is more focussed on reviewing the code, while I am trying to nudge the question towards including the human aspects too. In my head it's something like "I'm worried that I haven't understood all the edge cases in the interactions with ..." vs "I would like extra eyes on the handling of edge cases with ..." which tell me (as a reviewer) a different story

I don't feel strongly about this (and won't pretend to know the team dynamic well enough for a nuance like this anwyay), so unless you want to chat more I'll move along.

Reply:
I suppose it's fine as is as we're not looking for a direct answer per se but rather to nudge the author of the change to consider that aspect.